### PR TITLE
chore(linting): Trying to make eslint quieter

### PR DESCRIPTION
**Pull request by rusackas**
### SUMMARY

Disabling the rule about exhaustive hook dependencies, and just tweaking a bunch of other things to lessen the noise. 

Reduces the warnings from 275 down to 81.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by michael-s-molina — COMMENTED** [review]
@rusackas I think we shouldn't disable this rule as it's really important. Maybe make it as an error so it's not ignored anymore for new files. For current files, we could ignore the error for now and slowly fix them. It will require registering all current files that contain this error 😅

**Comment by rusackas** [comment]
> @rusackas I think we shouldn't disable this rule as it's really important. Maybe make it as an error so it's not ignored anymore for new files. For current files, we could ignore the error for now and slowly fix them. It will require registering all current files that contain this error 😅

Agreed. Sounds like the perfect thing to do over a beer some evening. It'll be a long, but simple PR. There are some other bits in this PR that might be worth looking at before I muddy the waters ;)

File: superset-frontend/packages/superset-ui-chart-controls/src/operators/sortOperator.ts [hunk 1]
```diff
@@ -28,42 +28,40 @@ import {
...
-    sortableLabels.includes(formData.x_axis_sort) &&
-    // the sort operator doesn't support sort-by multiple series.
-    isEmpty(formData.groupby)
-  ) {
-    if (formData.x_axis_sort === getXAxisLabel(formData)) {
       return {
         operation: 'sort',
         options: {
-          is_sort_index: true,
+          by: formData.x_axis_sort,
```
Comment (rusackas) [inline superset-frontend/packages/superset-ui-chart-controls/src/operators/sortOperator.ts:61]: Ummm.... I'm not sure how this got changed... it's certainly not a linting rule.

File: superset-frontend/packages/superset-ui-chart-controls/src/operators/sortOperator.ts [hunk 2]
```diff
@@ -28,42 +28,40 @@ import {
 import { PostProcessingFactory } from './types';
 import { extractExtraMetrics } from './utils';
 
-export const sortOperator: PostProcessingFactory<PostProcessingSort> = (
-  formData,
-  queryObject,
```
Comment (rusackas) [inline superset-frontend/packages/superset-ui-chart-controls/src/operators/sortOperator.ts:33]: I think this was unused, leading to all the reformatting here.

**Comment by rusackas** [comment]
Closing for now... I'll reconsider/rebuild this after we merge the eslint metrics and do some other things like whittling and bumping packages.

**Pull request closed without merging**